File: Algorithms/Data Structures/Hardwood Species.cpp
#include <iostream>
#include <map>
#include <string>

using namespace std;

map<string, int> trees;
map<string, int>::iterator it;

int processTrees(){
    string word;
    int count = 0;
    while(getline(cin, word) and word != ""){
        count++;
        if (trees.count(word) == 0){
            trees[word] = 1;
        }
        else{
            trees[word]++;
        }
    }
    return count;
}

int main(){
    int times;
    cin >> times;
    string jump;
    getline(cin, jump);
    getline(cin, jump);
    while(times--){
        int totalTrees = processTrees();
        for(it = trees.begin(); it!=trees.end(); it++){
            float freq = 100 * (float)it->second / (float)totalTrees;
            printf("%s %.4f\n", it->first.c_str(), freq);
        }
        trees.clear();
        if (times != 0) cout << endl;
    }
}

File: Algorithms/Data Structures/Hardwood Species.py
n = int(input())
input()
trees = {}
def processTree():
    try:
        count = 0
        while True:
            word = input()
            if word == "":
                return count
            if word not in trees:
                trees[word] = 1
            else:
                trees[word] += 1
            count += 1
    except EOFError:
        return count

while n:
    totalTrees = processTree()
    for a in trees.keys():
        print(a, " %.4f" % (trees[a] / totalTrees *100))
    trees = {}
    if n != 1: print()
    n-=1